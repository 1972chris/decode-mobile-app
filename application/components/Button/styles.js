import { StyleSheet } from 'react-native';

const button = {
  alignSelf: 'stretch',
  borderRadius: 2,
  elevation: 2,
  height: 36,
  marginBottom: 18,
  marginHorizontal: 16,
  marginTop: 10,
  paddingHorizontal: 16,
  paddingVertical: 8,
  shadowColor: 'rgba(0, 0, 0, 0.54)',
  shadowOffset: { width: 0, height: 2 },
  shadowOpacity: 0.8,
  shadowRadius: 2,
};

const buttonText = {
  alignSelf: 'center',
  fontSize: 16,

  fontWeight: '500',
};

const styles = StyleSheet.create({
  signButton: {
    ...button,
    backgroundColor: 'rgb(0,163,158)',
  },
  signButtonDisabled: {
    ...button,
    backgroundColor: 'rgba(0,163,158,0.4)',
  },
  buttonText: {
    ...buttonText,
    color: '#FFF',
  },
  disabledButtonText: {
    ...buttonText,
    color: 'rgba(255, 255, 255, 0.6)',
  },
});

export default styles;
