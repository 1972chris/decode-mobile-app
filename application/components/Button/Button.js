import React from 'react';
import { StyleSheet, Text, TouchableOpacity, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

function Button(props) {
  const buttonTextStyle = props.enabled ? styles.buttonText : styles.disabledButtonText;
  const buttonStyle = props.enabled ? styles.signButton : styles.signButtonDisabled;

  const style = StyleSheet.create(props.style);

  return (
    <TouchableOpacity
      disabled={!props.enabled}
      style={[buttonStyle, style]}
      onPress={props.onPress}
    >
      <Text style={buttonTextStyle}>{props.name}</Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  enabled: PropTypes.bool,
  onPress: PropTypes.func,
  name: PropTypes.string.isRequired,
  style: ViewPropTypes.style, // eslint-disable-line
};

Button.defaultProps = {
  onPress: () => {
  },
  enabled: true,
  style: {},
};

export default Button;

