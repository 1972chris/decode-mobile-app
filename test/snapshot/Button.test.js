import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../application/components/Button/Button';

it('renders Button enabled', () => {
  const rendered = renderer.create(<Button enabled name="" />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders Button disabled', () => {
  const rendered = renderer.create(<Button enabled={false} name="" />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders Button with correct name', () => {
  const rendered = renderer.create(<Button enabled name="A Name" />).toJSON();
  expect(rendered).toMatchSnapshot();
});
