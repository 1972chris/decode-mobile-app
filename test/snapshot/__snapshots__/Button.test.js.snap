// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button disabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "rgba(0,163,158,0.4)",
      "borderRadius": 2,
      "elevation": 2,
      "height": 36,
      "marginBottom": 18,
      "marginHorizontal": 16,
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "rgba(0, 0, 0, 0.54)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "rgba(255, 255, 255, 0.6)",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`renders Button enabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "rgb(0,163,158)",
      "borderRadius": 2,
      "elevation": 2,
      "height": 36,
      "marginBottom": 18,
      "marginHorizontal": 16,
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "rgba(0, 0, 0, 0.54)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`renders Button with correct name 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "rgb(0,163,158)",
      "borderRadius": 2,
      "elevation": 2,
      "height": 36,
      "marginBottom": 18,
      "marginHorizontal": 16,
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "rgba(0, 0, 0, 0.54)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    A Name
  </Text>
</View>
`;
